### ISSUE TITLE
Dependencies tree

### ISSUE THREAD

**Issue by idnovic** [X3n0m0rph59/precached#9]
Does precached generate a list of the most important files?
Lets say 10 often used programs use file a 9 times, file b 7 times, file c is only used by 2, file d is only used with/after file c....?

**Comment by X3n0m0rph59** [comment]
> Does precached generate a list of the most important files?

Yes, but there is room for improvement. Precached currently stores the count of executions of programs (represented by a histogram).

You may view this histogram with:

`$ sudo precachedctl plugins hot-applications list`

When precached performs offline-prefetching (ahead of time), it will replay the trace logs of the most important programs in the order of its execution count.

>Lets say 10 often used programs use file a 9 times, file b 7 times, file c is only used by 2, file d is only used with/after file c....?

What you describe resembles a so called Markov-chain. It is planned to implement such a Markov-chain based algorithm in the near future.

**Comment by idnovic** [comment]
Thank you for your time. I find precached very useful and interesting :)

**Comment by idnovic** [comment]
Cool also I found the optimize option.
Do I need to optimize regularly or is the optimization automated?

**Comment by X3n0m0rph59** [comment]
Optimization of trace logs is handled by precached automatically (during system idle time).
Manual optimization may be triggered (online) by issuing this command:

`$ sudo precachedctl housekeeping`

If online optimization is not desired, one can use:

`$ sudo iotracectl optimize`

The histogram does not take up much space, so the decision was made, to only optimize on user request!
